### PR TITLE
Hm/hybridize

### PR DESCRIPTION
**Pull request by haydenmcp**
Hybridized, adapted to Firefox.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -15,7 +16,7 @@
   "private": false,
   "repository": {
     "type": "git",
-    "url": "git://github.com/johnthad/paper-tristate-checkbox"
+    "url": "git://github.com/haydenmcp/paper-tristate-checkbox"
```
Comment (admwx7) [inline bower.json]: Base this against the hackception repo

File: bower.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "paper-tristate-checkbox",
-  "version": "0.1.0",
+  "version": "1.0.0",
```
Comment (admwx7) [inline bower.json]: Don't version bump in a feature PR

File: bower.json [hunk 3]
```diff
@@ -27,21 +28,45 @@
     "tests"
   ],
   "dependencies": {
-    "polymer": "Polymer/polymer#^1.3.0",
-    "iron-validatable-behavior": "PolymerElements/iron-validatable-behavior#^1.0.0",
-    "iron-form-element-behavior": "PolymerElements/iron-form-element-behavior#^1.0.0",
-    "paper-behaviors": "PolymerElements/paper-behaviors#^1.0.0",
-    "paper-ripple": "PolymerElements/paper-ripple#^1.0.0",
-    "paper-styles": "PolymerElements/paper-styles#^1.1.0"
+    "polymer": "Polymer/polymer#^1.0 || ^2.0",
```
Comment (admwx7) [inline bower.json]: Bump to 1.9 at the minimum

File: bower.json [hunk 4]
```diff
@@ -27,21 +28,45 @@
...
+    "iron-flex-layout": "PolymerElements/iron-flex-layout#^1.0 || ^2.0",
+    "iron-test-helpers": "PolymerElements/iron-test-helpers#^1.0 || ^2.0",
+    "web-component-tester": "Polymer/web-component-tester#^6.0.0",
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^1.0.0"
   },
-  "main": "paper-tristate-checkbox.html"
+  "variants": {
+    "1.x": {
+      "dependencies": {
+        "polymer": "Polymer/polymer#^1.9.3",
```
Comment (admwx7) [inline bower.json:48]: This needs to match the lower case of the or above

File: bower.json [hunk 5]
```diff
@@ -27,21 +28,45 @@
...
+      },
+      "devDependencies": {
+        "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
+        "web-component-tester": "^4.0.0",
+        "test-fixture": "PolymerElements/test-fixture#^1.0.0",
+        "iron-component-page": "PolymerElements/iron-component-page#^1.0.0",
+        "iron-demo-helpers": "PolymerElements/iron-demo-helpers#^1.0.0",
+        "iron-flex-layout": "PolymerElements/iron-flex-layout#^1.0.0",
+        "iron-test-helpers": "PolymerElements/iron-test-helpers#^1.0.0"
+      }
```
Comment (admwx7) [inline bower.json]: You need a resolution to force webcomponents in 1.x

File: paper-tristate-checkbox.html [hunk 6]
```diff
@@ -311,40 +305,49 @@
...
-          computed: '_computeValue(state)'
-        }
+          computed: '_computeValue(state)',
+        },
+      },
+
+      attached() {
+        this.setAttribute('aria-checked', 'mixed');
+        this._computeValue(this.state);
+        this.addEventListener('click', this._regularTap);
```
Comment (admwx7) [inline paper-tristate-checkbox.html]: You need a ready block to bind the function:
`this._regularTap = this._regularTap.bind(this);`

File: test/index.html [hunk 7]
```diff
@@ -8,6 +8,7 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
     <title>paper-tristate-checkbox tests</title>
+    <script src=../../webcomponentsjs/webcomponents-lite.js></script>
```
Comment (admwx7) [inline test/index.html]: Missing "

**Review by admwx7 — CHANGES_REQUESTED** [review]

File: test/index.html (hunk 7 above)
Comment (haydenmcp) [reply test/index.html]: done

File: bower.json (hunk 5 above)
Comment (haydenmcp) [reply bower.json]: done

File: bower.json (hunk 4 above)
Comment (haydenmcp) [reply bower.json:48]: done

File: bower.json [hunk 8]
```diff
@@ -27,21 +28,48 @@
...
+        "polymer": "Polymer/polymer#^1.9.3",
+        "iron-validatable-behavior": "PolymerElements/iron-validatable-behavior#^1.0.0",
+        "iron-form-element-behavior": "PolymerElements/iron-form-element-behavior#^1.0.0",
+        "paper-behaviors": "PolymerElements/paper-behaviors#^1.0.0",
+        "paper-ripple": "PolymerElements/paper-ripple#^1.0.0",
+        "paper-styles": "PolymerElements/paper-styles#^1.1.0"
+      },
+      "devDependencies": {
+        "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
+        "web-component-tester": "^4.0.0",
```
Comment (stramel) [inline bower.json:57]: `Polymer/web-component-tester#^5.0.0`

File: bower.json [hunk 9]
```diff
@@ -27,21 +28,48 @@
...
+        "iron-flex-layout": "PolymerElements/iron-flex-layout#^1.0.0",
+        "iron-test-helpers": "PolymerElements/iron-test-helpers#^1.0.0"
+      },
+      "resolutions": {
+        "webcomponentsjs": "^0.7.0"
+      }
+    }
+  },
+  "resolutions": {
+    "test-fixture": "2.0.1",
```
Comment (stramel) [inline bower.json]: Is this necessary?

File: paper-tristate-checkbox.html [hunk 10]
```diff
@@ -311,40 +305,51 @@
...
          */
         value: {
           type: String,
-          computed: '_computeValue(state)'
-        }
+          computed: '_computeValue(state)',
+        },
+      },
+      ready() {
+        this._regularTap = this._regularTap.bind(this);
```
Comment (stramel) [inline paper-tristate-checkbox.html]: `this._regularClick` instead of `this._regularTap`

File: paper-tristate-checkbox.html [hunk 11]
```diff
@@ -311,40 +305,51 @@
...
+      },
+      attached() {
+        this.setAttribute('aria-checked', 'mixed');
```
Comment (stramel) [inline paper-tristate-checkbox.html]: Is there a reason this isn't working in the host attributes?

File: test/index.html [hunk 12]
```diff
@@ -8,13 +8,14 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
     <title>paper-tristate-checkbox tests</title>
+    <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
     <script src="../../web-component-tester/browser.js"></script>
   </head>
   <body>
     <script>
       WCT.loadSuites([
         'basic.html',
```
Comment (stramel) [inline test/index.html]:
```
'basic.html?dom=shadow',
'basic.html?wc-shadydom=true&wc-ce=true',
```

**Review by stramel — APPROVED** [review]

File: paper-tristate-checkbox.html (hunk 11 above)
Comment (haydenmcp) [reply paper-tristate-checkbox.html]: Yeah, it's interesting. A behavior dependency has itself another dependency which listeners to changes to an ariaActiveAttribute. As a result, a distant observer is triggered which removes the attribute so it stopped showing up in 2.x. The dependency is PaperInkyFocusBehavior which has a dependency on IronButtonState. This was an interesting bug to tackle.

File: bower.json [hunk 13]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "paper-tristate-checkbox",
-  "version": "0.1.0",
+  "version": "0.3.0",
```
Comment (admwx7) [inline bower.json:3]: Change this to 0.2.0, when we release we'll do a version bump on this file as well.

File: bower.json (hunk 4 above)
Comment (admwx7) [reply bower.json:48]: Meaning if this is dependent on ^1.9.3 it means it should have the same requirement above.

**Review by admwx7 — APPROVED** [review]

**Pull request merged**